### PR TITLE
3.2 release of pgt

### PR DESCRIPTION
**Pull request by bgruening**
This is a major release with a lot of new features but also some deprecations in preparation for a new 4.0 release.

During the years we have introduced several ways to enable/disable settings.
We have used on/off, yes/no, 0/1, true/false. From now on we recommend to only use true/false so that we can unify our config files and make it more intuitive for all pgt-users.

We also removed all 2/3/4 words items and concatenate them now by `_` (https://github.com/deeptools/pyGenomeTracks/pull/132).
For example `line width` is now `line_width`. Thanks @LeilyR!

A few new features in this release:
* Every config is now checked for syntax errors before anything is executed (@lldelisle)
* Added the possibility to merge transcripts into one single gene representation when using gtf (@lldelisle)
* Added the possibility to rasterize bedgraph plots (@lldelisle)
* Added the possibility to use summary functions on bedgraphs (@lldelisle)
* Generate an empty track if a requested region is not in the given track-files. Fixed https://github.com/deeptools/pyGenomeTracks/issues/120 (@LeilyR)
* Generate an empty track if a chromosome is missing in bedgraph files (@lldelisle)
* Improved UCSC style for intron arrows (@lldelisle)
* Flybase style now supports color_utr and height_uts (@lldelisle)
* A new tracktype `hlines` was added (@lldelisle)
* Allow to plot the arcs of the links with a color scale based on scores as proposed in https://github.com/deeptools/pyGenomeTracks/issues/30 (@lldelisle)
* Allow to plot rectangle on a heatmap for loops. Fixed https://github.com/deeptools/pyGenomeTracks/issues/47 (@Phlya, @lldelisle)
* A lot of HiC Matrix improvements (@lldelisle)
* The `alpha` property can now be used nearly everywhere (@lldelisle)

Also checkout our updated [readme](https://github.com/deeptools/pyGenomeTracks/blob/master/README.md).

Special thanks to @lldelisle & @LeilyR for such a feature-rich release.

Merry Xmas and a happy new year!
Your PGT team!

### CODE REVIEW

File: pygenometracks/tests/test_data/epilogos.ini [hunk 1]
```diff
@@ -2,24 +2,24 @@
...
 file = epilog.qcat.bgz
 height = 5
-title = height=5
-orientation=inverted
+title = height = 5
+orientation = inverted
 
 [spacer]
-width=0.5
+height=0.5
```
Comment (lldelisle) [inline pygenometracks/tests/test_data/epilogos.ini]: maybe put spaces around `=`.

File: pygenometracks/tests/test_data/master_tracks.ini [hunk 2]
```diff
@@ -87,62 +99,87 @@ height = 2
...
+#max_labels = 60
 # optional: font size can be given to override the default size
 fontsize = 10
-# optional: line width
-#line width = 0.5
+# optional: line_width
+#line_width = 0.5
 # the display parameter defines how the bed file is plotted.
-# The options are ['colapsed', 'interleaved', 'triangles'] This options asume that the regions do not overlap.
+# The options are ['collapsed', 'interleaved', 'triangles'] This options asume that the regions do not overlap.
```
Comment (lldelisle) [inline pygenometracks/tests/test_data/master_tracks.ini]: These options

**Comment by bgruening** [comment]
Done. Thanks!

File: environment.yml [hunk 3]
```diff
@@ -7,7 +7,7 @@ channels:
    - conda-forge
 dependencies:
     - numpy >=1.16
-    - matplotlib >=3.0
+    - matplotlib =3.1.1
```
[the remark this answers is not in the document]
Comment (LeilyR) [reply environment.yml]: no, i did it on purpose, it changed pixels and some tests was failed.

**Comment by lldelisle** [comment]
You need to rerun:
```
make_tracks_file --trackFiles pygenometracks/tests/test_data/Li_et_al_2015.h5 pygenometracks/tests/test_data/bigwig_chrx_2e6_5e6.bw pygenometracks/tests/test_data/tad_classification.bed pygenometracks/tests/test_data/epilog.qcat.bgz -o pygenometracks/tests/test_data/master_tracks.ini
```

**Comment by bgruening** [comment]
Thanks @lldelisle ... I should not try to do this during work hours these days.

**Pull request merged**